### PR TITLE
Update LLVM submodule

### PR DESCRIPTION
**Pull request by youngar**
This is quite a large change.  Separated into commits to make it easier to review.

1. Update LLVM submodule

2. Update circt-translate help message

New version of MLIR properly prints the help message.  This updates the
command line test for it.

3. Do not use methods from OpState that just forward to Operation

These methods are being removed from mlir::OpState in reviews.llvm.org/D94191

4. Use standard integer types in LLHDToLLVM

The type system for the LLVM dialect is undergoing changes to use types
from the standard dialect. This includes the removal of LLVM integer
types, removal of the common parent type LLVMType, new helpers for
working with LLVM structs, arrays, and pointers.

for more information see:
reviews.llvm.org/D94178
reviews.llvm.org/D93680
reviews.llvm.org/D93713

### CODE REVIEW

**Comment by youngar** [comment]
I think I need to bump the https://github.com/circt/llvm fork.  PR here: https://github.com/circt/llvm/pull/6

File: include/circt/Dialect/RTL/RTLCombinatorial.td [hunk 1]
```diff
@@ -46,7 +46,7 @@ def ConstantOp
     OpBuilderDAG<(ins "int64_t":$value, "IntegerType":$type)>
   ];
   let extraClassDeclaration = [{
-    APInt getValue() { return getAttrOfType<IntegerAttr>("value").getValue(); }
+    APInt getValue() { return (*this)->getAttrOfType<IntegerAttr>("value").getValue(); }
```
Comment (lattner) [inline include/circt/Dialect/RTL/RTLCombinatorial.td:49]: Plz check this for 80 columns.

File: include/circt/Dialect/RTL/RTLStructure.td [hunk 2]
```diff
@@ -46,7 +46,7 @@ def RTLModuleOp : RTLOp<"module",
 
     // Get the module's symbolic name.
     StringRef getName() {
-      return getAttrOfType<StringAttr>(::mlir::SymbolTable::getSymbolAttrName())
+      return (*this)->getAttrOfType<StringAttr>(::mlir::SymbolTable::getSymbolAttrName())
```
Comment (lattner) [inline include/circt/Dialect/RTL/RTLStructure.td:49]: plz check for 80 cols

**Review by lattner — APPROVED** [review]
Awesome, thank you for doing this!

**Review by teqdruid — APPROVED** [review]

**Pull request merged**